### DISCUSSION TITLE
I don't know how to ref the component

### DISCUSSION THREAD

**Discussion by jinwoo-web** [nuxt/nuxt#16612] [Questions] [answered]
I can't component dynamic ref.
please see my very simple code

https://stackblitz.com/edit/github-abtcoj-d5aias?file=app.vue

**I want to access test variable inside modal**

**Comment by TheAlexLichter** [comment]
I am not sure what exactly your problem. Could you try describing it again?

**Reply by jinwoo-web** [reply]
I want to access the test variable inside the Modal.vue component.
In vue3, using ref Variables can be accessed within the component.
Is there any other way for nuxt3?

**Reply by jinwoo-web** [reply]
```
<script lang="ts" setup>
const modal = ref();
function test() {
  console.log(modal); 
  // why is the modal null
}
</script>
<template>
  <div>
    <Modal ref="modal" />
    <button @click="test">click</button>
  </div>
</template>
```

**Reply by TheAlexLichter** [reply]
`modal.value` is not null (after the component has been mounted) and shouldn't be when clicking the button

**Comment by jinwoo-web** [comment]
I found the cause of the problem

```
-- Modal.vue
<script lang="ts">
export default {
  setup() {
    const test = () => {};

    return {
      test,
    };
  },
};
</script>
```
'test' function can be called from parent component

```
-- Modal.vue
<script lang="ts" setup>
const test = () => {};
</script>
```
Cannot call 'test' function from parent component


Is this the intended behavior in Nuxt3?

**Comment by TheAlexLichter** [comment, marked as answer]
Ahh, you want to access the `test` variable of `Modal` in a parent component? Then check out [`defineExpose`](https://vuejs.org/api/sfc-script-setup.html#defineexpose) for `<script setup>`. See also [this SO question](https://stackoverflow.com/a/73567049/3975480).

**Reply by jinwoo-web** [reply]
oh~ thank you
it wasn't a problem nuxt3
